src/Engines/engine.ts

```diff
@@ -290,6 +291,39 @@ export class Engine extends ThinEngine {
         return EngineStore.LastCreatedScene;
     }
 
+    /**
+     * Engine abstraction for createImageBitmap
+     * @param args parameters for createImageBitmap implementation
+     * @returns ImageBitmap
+     */
+    public createImageBitmap(...args: any[]): Promise<ImageBitmap>

```

Comment:
Why does this have to be `any[]` for parameters? I think there is something fundamental about this API that I'm not understanding.

Reply:
createImageBitmap has 2 signatures : https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/createImageBitmap
So, aby number of parameters are supported here.

Reply:
Hmm, having two signatures doesn't imply that you have to use any or `...`. You can just create overloads in TypeScript (as long as there are different number of arguments).

Reply:
From lib.dom.d.ts:

```typescript
declare function createImageBitmap(image: ImageBitmapSource, options?: ImageBitmapOptions): Promise<ImageBitmap>;
declare function createImageBitmap(image: ImageBitmapSource, sx: number, sy: number, sw: number, sh: number, options?: ImageBitmapOptions): Promise<ImageBitmap>;
```

Reply:
from what I understand, you can declare overloadings but you can't have different signature for the implementation. I case either check parameters types for choosing the correct overloading or use `any` with `apply`. Which one shall I do?

Reply:
https://stackoverflow.com/questions/12688275/is-there-a-way-to-do-method-overloading-in-typescript

Here is an example: https://www.typescriptlang.org/play?#code/MYGwhgzhAEBiD29oG8CwAoaXoAcCuARiAJbDQBCYATgBRgBc0AdngLYECmVANNAYy3ZcAlIwBu8YgBMA3Bmy5CJMpVoNoEAC5ViTAOajoE6XMzZ8RUhWp0BbTlWgAfDdt17eBAPx2hVQ8ZSKPLYAL4YoUA

Reply:
I was taking the problem from the wrong side. there is only 1 override used in whole engine. So I changed the signature to only support it.